app/code/Magento/Sales/Controller/Adminhtml/Order/Invoice/AddComment.php

```diff
@@ -90,7 +96,7 @@ public function execute()
             );
 
             $this->invoiceCommentSender->send($invoice, !empty($data['is_customer_notified']), $data['comment']);
-            $invoice->save();
+            $this->invoiceRepository->save($invoice);
```

Comment:
$this->invoiceRepository is not declared. Please add and declare it in BC way as you did it at Magento/Sales/Controller/Adminhtml/Invoice/AbstractInvoice/View.php

Reply:
I'm passing `$invoiceRepository` to the parent just like `$context`

Reply:
@drpayyne 
But what will be if somebody has already overridden Magento\Sales\Controller\Adminhtml\Invoice\AbstractInvoice\View? 
$this->invoiceRepository is optional and could be not instantiated at all with the old implementation of Magento\Sales\Controller\Adminhtml\Invoice\AbstractInvoice\View.

Reply:
@Stepa4man If the optional parameter isn’t passed in already, it defaults to null and passes null to parent, which initializes the object in constructor since null was received. The parent has the variable declared. I’m not sure why the variable has to be declared in child too. Just like `$context` isn’t.